tests/testthat/test-tokens_xptr.R

```diff
@@ -312,3 +312,43 @@ test_that("lengths works on tokens xptr objects", {
         NULL
     )
 })
+
+test_that("[[ indexing works on tokens_xptr", {
+    toks <- tokens(c("A a a, b.", "A b c c c?"))
+    xtoks <- as.tokens_xptr(toks)
+    expect_identical(
+        toks[[2]],
+        xtoks[[2]]
+    )
+})
+    
+test_that("n functions works the same on both tokens and tokens_xptr", {
+    toks <- tokens(c("A a a, b.", "A b c c c?"))
+    xtoks <- as.tokens_xptr(toks)
+    expect_identical(
+        ntoken(toks), 
+        ntoken(xtoks)
+    )
+    expect_identical(
+        ntoken(toks, remove_punct = TRUE), 
+        ntoken(xtoks, remove_punct = TRUE)
+    )
+    # check that the former removals did not modify xtoks
+    expect_identical(
```

Comment:
This second check should not fail - but does if the former call is used without the x2 deep copies. But @koheiw you might know a better way. I suggest you try improving this.